core/src/edu/cornell/gdiac/main/LevelMode.java

```diff
@@ -0,0 +1,162 @@
+package edu.cornell.gdiac.main;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton.TextButtonStyle;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+import edu.cornell.gdiac.util.ScreenListener;
+
+/**
+ * Level selection screen.
+ */
+public class LevelMode implements Screen {
+
+  /**
+   * Exit code for quitting the game
+   */
+  public static final int EXIT_QUIT = 0;
+  /**
+   * Code for changing to level menu
+   */
+  public static final int EXIT_TO_GAME = 1;
+  /**
+   * Listener that will update the player mode when we are done
+   */
+  private ScreenListener listener;
+
+  /**
+   * The selected level asset key
+   */
+  private String selectedLevelKey;
+
+  /**
+   * Reference to GameCanvas created by the root
+   */
+  private GameCanvas canvas;
+
+  /**
+   * Whether this is an active controller
+   */
+  private boolean active;
+
+  /**
+   * Levels stage
+   */
+  private Stage levelsMap;
+
+  public LevelMode(String[] levelKeys, GameCanvas canvas) {
+    this.canvas = canvas;
+    canvas.clear();
+
+    active = false;
+
+    // Representing the levels selection map
+    levelsMap = new Stage();
+
+    Table table = new Table();
+    table.setFillParent(true);
+    // Create three equally spaced columns
+    table.defaults().expandX().uniformX();
```

Comment:
where does the 3 number come from here? you're code obviously works but I'm just wondering how lol. Also if we wanted to add more levels would the UI be able to adapt easily?